File: 백준/Silver/10828. 스택/스택.py
import sys
def push(x):
    stack.append(x)
def pop():
    if len(stack) == 0:
        print(-1)
    else:
        top = stack.pop()
        print(top)
def size():
    print(len(stack))
def empty():
    if len(stack) == 0:
        print(1)
    else:
        print(0)
def top():
    if len(stack) == 0:
        print(-1)
    else:
        print(stack[-1])
N = int(sys.stdin.readline())
stack = []
for _ in range(N):
    word = sys.stdin.readline().split()
    order = word[0]
    if order == "push":
        push(word[1])
    elif order == "pop":
        pop()
    elif order == "size":
        size()
    elif order == "empty":
        empty()
    elif order == "top":
        top()

File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Stack;

public class Main {

	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
		int N = Integer.parseInt(br.readLine());
		Stack<Integer> stack = new Stack<>();

		for (int i = 0; i < N; i++) {
			String[] command2 = br.readLine().split(" ");
			
			switch (command2[0]) {
			case "push":
				stack.push(Integer.parseInt(command2[1]));
				break;
			case "pop":
				if (stack.empty()) {
					bw.write(-1 + "\n");
				} else {
					bw.write(stack.pop() + "\n");
				}
				break;
			case "size":
				bw.write(stack.size() + "\n");
				break;
			case "empty":
				if (stack.empty()) {
					bw.write(1 + "\n");
				} else {
					bw.write(0 + "\n");
				}
				break;
			case "top":
				if (stack.empty()) {
					bw.write(-1 + "\n");
				} else {
					bw.write(stack.peek() + "\n");
				}
				break;
			default:
				break;
			}
			
			
		}
		
		bw.flush();
		bw.close();
		br.close();
	}

}